pkg/scheduler/algorithm/priorities/balanced_resource_allocation_test.go

```diff
@@ -17,17 +17,90 @@ limitations under the License.
 package priorities
 
 import (
+	"fmt"
 	"reflect"
 	"testing"
 
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/kubernetes/pkg/features"
 	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
 	"k8s.io/kubernetes/pkg/scheduler/schedulercache"
 )
 
+// getExistingVolumeCountForNode gets the current number of volumes on node.
+func getExistingVolumeCountForNode(pods []*v1.Pod, maxVolumes int) int {
+	volumeCount := 0
+	for _, pod := range pods {
+		volumeCount += len(pod.Spec.Volumes)
```

Comment:
Can't we have shared volumes?

cc/ @msau42

Reply:
Do you care about counting bind mounts?  There is always a bind-mount (done by the container runtime) for all the volumes referenced in the container spec.

At the pod level, you can share between pods, but the number of mounts depends on the volume type.  Sharing between pods on the same node is not as common.  Typically, scheduler tries to spread pods across nodes.
- In the case of NFS, each pod has its own NFS mount.
- For something like GCE PD (sharing should be a rare use case), you will have only one global node mount, but a bind mount per pod.
- Volume types like EmptyDir, ConfigMap, Secrets, downward api are not mounted 
- Hostpath volumes can be mounted or not, we can't tell from the spec.

Reply:
This is a function used in testing, thats why I did not bother checking that, if you think its important, I can add a check.

Reply:
Thanks Ravi for pointing out. I missed the fact that it is a test only function. We don't need to change it.